### PR TITLE
[Editor] Add few more info when saving ink data (thickness, opacity, …)

### PR DESCRIPTION
**Pull request by calixteman**
Fix the InkList entry: the coordinates were relative to the page and not to the bounding box of the annotation.

### CODE REVIEW

**Comment by calixteman** [comment]
I must update the coordinates in `test_manifest.json`.

File: src/core/annotation.js [hunk 1]
```diff
@@ -4070,6 +4070,20 @@ class InkAnnotation extends MarkupAnnotation {
     ink.set("Border", [0, 0, 0]);
```
Comment (Snuffleupagus) [inline src/core/annotation.js:4070]: I don't think this line makes sense now that you set a `BS`-entry below, since in our own code we ignore such an entry in that case; note https://github.com/mozilla/pdf.js/blob/071e6bc7e77754d9f1f13a84a05851d23a657971/src/core/annotation.js#L797-L835

File: src/core/annotation.js [hunk 2]
```diff
@@ -4070,6 +4070,20 @@ class InkAnnotation extends MarkupAnnotation {
...
     ink.set("Rotate", rotation);
 
+    // Line thickness.
+    const bs = new Dict(xref);
+    ink.set("BS", bs);
+    bs.set("W", thickness);
```
Comment (Snuffleupagus) [inline src/core/annotation.js:4076]: If you're adding a `BS`-dict, let's give it a proper /Type-entry as well.
```suggestion
    // Line thickness.
    const bs = new Dict(xref);
    ink.set("BS", bs);
    bs.set("Type", Name.get("Border"));
    bs.set("W", thickness);
```

Comment (calixteman) [reply src/core/annotation.js:4076]: I wondered the same and I checked how Acrobat is saving the BS and it doesn't add a Type for the ink case.

**Comment by calixteman** [comment]
The coordinates in the ink list are wrong when the annotation is rotated: I'll fix that tomorrow.

**Comment by calixteman** [comment]
The test `tracemonkey-editor-rotation` will fail but it's fine: I changed the way to compute the coordinates in order to have correct InkList so I recreate some new coordinates for this test.

**Comment by calixteman** [comment]
I checked that the `InkList` are correct in Acrobat: when editing a curve this one is regenerated from the list, hence it's easy to see if the coordinates are correct.

File: test/test_manifest.json
Comment (Snuffleupagus) [inline test/test_manifest.json:1]: There seem to be *a lot* of unrelated whitespace changes in this file now, please revert that.

Comment (calixteman) [reply test/test_manifest.json:1]: Aarf... it's painful.
Could we format it with `gulp lint` and get rid of these kind of issues ?

File: src/display/editor/ink.js [hunk 3]
```diff
@@ -910,137 +910,123 @@ class InkEditor extends AnnotationEditor {
...
+    return points;
+  }
+
   /**
    * Transform and serialize the paths.
    * @param {number} s - scale factor
    * @param {number} tx - abscissa of the translation
    * @param {number} ty - ordinate of the translation
-   * @param {number} h - height of the bounding box
+   * @param {Array<number>} rect - the rectangle cotaining the paths
```
Comment (Snuffleupagus) [inline src/display/editor/ink.js:990]: Please fix the `cotaining` -> `containing` spelling.
Also, isn't the comment slightly misleading now since as far as I can tell `rect` is the bounding box (and doesn't contain any paths)?

**Review by Snuffleupagus — APPROVED** [review]
r=me, with the comment fixed; thank you!
